lib/rw-overlay.nix

```diff
@@ -1,28 +1,28 @@
 #needs to be a runtime eval
-{writeShellScriptBin,
+{writeShellScriptBin, fuse-overlayfs,
 mountprefix, callback, base}:
   writeShellScriptBin "mounter" ''
     set -euo pipefail #http://redsymbol.net/articles/unofficial-bash-strict-mode/
     export SHELLOPTS #Note export exports the key not the value, "updates propagate"
     set -x
 
     mount="${mountprefix}"
+    gio trash "$mount" || true
     mkdir -p "$mount" || true #TODO
 
     pushd "$mount"
     #TODO do i actually need all of these?
-    mkdir mutable upper weirdsrc work bind || true #TODO
-    #TODO other FS options? is there any way to not need root?
-    sudo mount -t overlay weirdsrc -o lowerdir="${base}",upperdir=upper,workdir=work mutable #TODO? , metacopy=on /check other opts
-    #these are an adhoc mess figure them out
-    sudo chown -R $USER:100 mutable
-    sudo chmod -R gu+w mutable
-    sudo chmod -R 777 work/work #TODO wat
+    mkdir mutable upper work bind || true #TODO
+
+    cp -r "${base}" lower
+    chmod -R gu+w lower
```

Comment:
I do not like this. It would be nice to be able to mount the lower layer as “writeable” so that any changes to the existing files would be copied to the upper layer on write but I could not find anything like that.

Reply:
Maybe a plug-in like https://github.com/containers/fuse-overlayfs/pull/119 could do that.